snorkel/labeling/preprocess/core.py

```diff
@@ -85,35 +85,33 @@ def __call__(self, x: DataPoint) -> DataPoint:
 
 
 class LambdaPreprocessor(Preprocessor):
-    def __init__(
-        self,
-        f: Callable[..., FieldMap],
-        field_names: Mapping[str, str],
-        preprocessed_field_names: Mapping[str, str],
-        mode: PreprocessorMode = PreprocessorMode.NONE,
-    ) -> None:
-        super().__init__(field_names, preprocessed_field_names, mode)
+    def __init__(self, f: Callable[..., FieldMap]) -> None:
+        """Convenience class for Preprocessors that execute a simple
+        function with no set up. The function arguments are parsed
+        to determine the input field names of the data points.
+
+        Args:
+            * f: the function executing the preprocessing operation
+        """
         self._f = f
+        field_names = {k: k for k in inspect.getfullargspec(f)[0]}
+        super().__init__(field_names=field_names, preprocessed_field_names=None)
 
     def preprocess(self, **kwargs: Any) -> FieldMap:
         return self._f(**kwargs)
 
 
-class preprocessor:
-    def __init__(
-        self,
-        field_names: Mapping[str, str],
-        preprocessed_field_names: Mapping[str, str],
-        mode: PreprocessorMode = PreprocessorMode.NONE,
-    ) -> None:
-        self.field_names = field_names
-        self.preprocessed_field_names = preprocessed_field_names
-        self.mode = mode
+def preprocessor(f: Callable[..., FieldMap]) -> LambdaPreprocessor:
+    """Decorator to define a LambdaPreprocessor object from a function
+
+        Example usage:
 
-    def __call__(self, f: Callable[..., FieldMap]) -> LambdaPreprocessor:
-        return LambdaPreprocessor(
-            f=f,
-            field_names=self.field_names,
-            preprocessed_field_names=self.preprocessed_field_names,
-            mode=self.mode,
-        )
+        ```
+        @preprocessor()
+        def concatenate_text(title: str, body: str) -> FieldMap:
+            return f"{title} {body}"
```

Comment:
I guess I commented on PRs in wrong order... this example should return a dict right?

Reply:
Yes correct, will update